android/src/main/java/io/wazo/callkeep/VoiceConnectionService.java

```diff
@@ -328,9 +330,39 @@ private Connection createConnection(ConnectionRequest request) {
 
         Bundle extras = request.getExtras();
         HashMap<String, String> extrasMap = this.bundleToMap(extras);
-        extrasMap.put(EXTRA_CALL_NUMBER, request.getAddress().toString());
+
+        String callerNumber = request.getAddress().toString();
+        if (callerNumber.contains(":")) {
+            //CallerNumber contains a schema which we'll separate out
+            int schemaIndex = callerNumber.indexOf(":");
+            String number = callerNumber.substring(schemaIndex + 1);
+            String schema = callerNumber.substring(0, schemaIndex);
+
+            extrasMap.put(EXTRA_CALL_NUMBER, number);
+            extrasMap.put(EXTRA_CALL_NUMBER_SCHEMA, schema);
+        }
+        else {
+            extrasMap.put(EXTRA_CALL_NUMBER, callerNumber);
+        }
```

Comment:
@manuquentin 

This is how I chose to address the schema thing you pointed out. This will at least ensure that it's always applied if schema should make it into the system elsewhere... Thoughts?

Reply:
👌